src/snarky.d.ts

```diff
@@ -98,60 +290,148 @@ export class Field {
+  */
 
-  static toJSON(x: Field): JSONValue;
-  static fromJSON(x: JSONValue): Field | null;
+  // static toJSON(x: Field): JSONValue;
 
-  static toConstant(x: Field): Field;
+  static fromJSON(x: JSONValue): Field | null;
 }
 
+/**
+ * An element of a finite field.
+ */
+export function Field(x: number | string): Field;
+
+/**
+ * A boolean value. You can use it like this:
+ *
+ * ```
+ * const x = new Bool(true);
+ * ```
+ *
+ * You can also combine multiple booleans via [[`not`]], [[`and`]], [[`or`]].
+ *
+ * Use [[assertEquals]] to enforce the value of a Bool.
+ */
+export function Bool(x: Bool | boolean): Bool;
 export class Bool {
   constructor(x: Bool | boolean);
 
+  /**
+   * Converts a [[`Bool`]] to a [[`Field`]]. `false` becomes 0 and `true` becomes 1.
+   */
   toField(): Field;
 
+  /**
+   * @returns a new [[`Bool`]] that is the negation of this [[`Bool`]].
+   */
   not(): Bool;
+
+  /**
+   * @param y A [[`Bool`]] to AND with this [[`Bool`]].
+   * @returns a new [[`Bool`]] that is set to true only if
+   * this [[`Bool`]] and `y` are also true.
+   */
   and(y: Bool | boolean): Bool;
+
+  /**
+   * @param y a [[`Bool`]] to OR with this [[`Bool`]].
+   * @returns a new [[`Bool`]] that is set to true if either
+   * this [[`Bool`]] or `y` is true.
+   */
   or(y: Bool | boolean): Bool;
 
+  /**
+   * Aborts the program if this [[`Bool`]] is equal to `y`.
+   * @param y a [[`Bool`]].
+   */
   assertEquals(y: Bool | boolean): void;
 
+  /**
+   * Returns true if this [[`Bool`]] is equal to `y`.
+   * @param y a [[`Bool`]].
+   */
   equals(y: Bool | boolean): Bool;
+
+  /**
+   * Returns true if this [[`Bool`]] is true.
+   */
+  // TODO: that seems useless :D?
   isTrue(): Bool;
+
+  /**
+   * Returns true if this [[`Bool`]] is false.
+   */
+  // not very useful no?
   isFalse(): Bool;
 
-  sizeInFieldElements(): number;
-  toFieldElements(): Field[];
+  sizeInFields(): number;
+  toFields(): Field[];
 
+  /**
+   * Serialize the [[`Bool`]] to a string, e.g. for printing.
+   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
+   */
   toString(): string;
+  /**
+   * Serialize the [[`Bool`]] to a JSON string.
+   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
+   */
   toJSON(): JSONValue;
 
-  /* Can only be called on non-witness values */
+  /**
+   * This converts the [[`Bool`]] to a javascript [[boolean]].
+   * This can only be called on non-witness values.
+   */
   toBoolean(): boolean;
 
   /* static members */
   static toField(x: Bool | boolean): Field;
 
   static Unsafe: {
+    /**
+     * Converts a [[`Field`]] into a [[`Bool`]]. This is a **dangerous** operation
+     * as it assumes that the field element is either 1 or 0
+     * (which might not be true).
+     * @param x a [[`Field`]]
+     */
     ofField(x: Field | number | string | boolean): Bool;
```

Comment:
Instead of creating this `Unsafe` thing, it seems better to just have a function that behaves like
`ofField: (x: Field) => x.equals(1)`